fix(android)(9_0_X): module build to download ndk with gradle tool 3.5.0+

jquick-axway opened the pull request:
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-27798

**Cherry-pick of PR:** #11613

jquick-axway committed:
fix(android): module build to download ndk with gradle tool 3.5.0+

- After updating Android gradle tool from 3.4.x to 3.6.0, module builds no longer auto-downloads NDK if missing.
- Issue on Google's end which default to an "unstable" NDK version and tool does not support downloading unstable versions.

build commented:
<!--
  0 failure: 
  1 warning:  This PR has miles...
  2 messages
  
  DangerID: danger-id-Danger;
-->


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>This PR has milestone set to 9.0.2, but the version defined in package.json is 9.0.1
Please either:
- Update the milestone on the PR
- Update the version in package.json
- Hold the PR to be merged later after a release and version bump on this branch</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>:fist: The commits in this PR match our conventions! Feel free to Rebase and Merge this PR when ready.</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>

  :white_check_mark: All tests are passing
Nice one! All 4490 tests are passing.
(There are 458 skipped tests not included in that total)
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 27c52263172da46d1723d52b9c8bbc3d4301aaeb
</p>

garymathews reviewed (approved):
CR: **PASS**

jquick-axway committed:
chore(android): default to older ndk version for modules


- Downgraded default NDK version for modules from 21.0.6113669 to 20.1.5948944
  * Avoids harmless "bad file descriptor" errors that get logged by newest NDK.

sgtcoolguy committed:
Merge branch '9_0_X' into TIMOB-27798-rev1-9_0_X

sgtcoolguy committed:
Merge branch '9_0_X' into TIMOB-27798-rev1-9_0_X

ssekhri commented:
FR Passed.

1. With no NDK on the system, the module build installs side-by-side NDK
2. When side-by-side NDK (any other version) is already installed the module build does not install the NDK
3. When side-by-side NDK is not installed but another NDK is on the system. "appc ti config" does not have flag for android.ndkPath set then the build for module installs side-by-side NDK.
4. When side-by-side NDK is not installed but another NDK is on the system. "appc ti config" flag for android.ndkPath is set to the NDK on the system then the module build does not install side-by-side NDK.

Verified on:
Mac OS: 10.15.4
SDK: 9.0.2.v20200429095029
Appc CLI: 8.0.0
JDK: 13.0.2
Node: 10.17.0
